response.go

```diff
@@ -179,34 +160,228 @@ func (c *CountResultItem) String() string {
 	return fmt.Sprintf("%d:%d", c.ID, c.Count)
 }
 
-func countItemsFromInternal(items []*pbuf.Pair) []*CountResultItem {
+func countItemsFromInternal(items []*pbuf.Pair) TopNResult {
 	result := make([]*CountResultItem, 0, len(items))
 	for _, v := range items {
 		result = append(result, &CountResultItem{ID: v.ID, Key: v.Key, Count: v.Count})
 	}
-	return result
+	return TopNResult(result)
+}
+
+type TopNResult []*CountResultItem
+
+func (TopNResult) Type() uint32 {
+	return QueryResultTypePairs
+}
+
+func (TopNResult) Bitmap() BitmapResult {
+	return defaultBitmapResult()
+}
+
+func (t TopNResult) CountItems() []*CountResultItem {
+	return t
+}
+
+func (TopNResult) Count() int64 {
+	return 0
+}
+
+func (TopNResult) Sum() int64 {
+	return 0
+}
+
+func (TopNResult) Changed() bool {
+	return false
 }
 
 // BitmapResult represents a result from Bitmap, Union, Intersect, Difference and Range PQL calls.
 type BitmapResult struct {
-	Attributes map[string]interface{}
-	Bits       []uint64
-	Keys       []string
+	Attributes map[string]interface{} `json:"attrs"`
+	Bits       []uint64               `json:"bits"`
+	Keys       []string               `json:"keys"`
 }
 
 func newBitmapResultFromInternal(bitmap *pbuf.Bitmap) (*BitmapResult, error) {
 	attrs, err := convertInternalAttrsToMap(bitmap.Attrs)
 	if err != nil {
 		return nil, err
 	}
+	bits := bitmap.Bits
+	if bits == nil {
+		bits = []uint64{}
```

Comment:
why can't we just leave it nil? Iteration over a nil slice works, len() of a nil slice works, anything accessing a member directly would need to check length first anyway.

Reply:
if you `json.Marshal` them they become `null` instead of `[]`.

Reply:
I think omitempty will take care of that

Reply:
`{}` is not the same as `{bits:[], attrs:[]}`.

Reply:
you're right, hmmm, I was hoping it would treat nil as empty and an empty list as an empty list, but no such luck. 

I'd be in favor of writing a custom MarshalJSON for Bitmap that sets the  defaults before Marshaling - that way we're only setting them in one place, and only if JSON is actually being used. Otherwise we won't incur the extra allocations. We can also handle the sticky business of ensuring that we always have "bits" or "keys", but not both (except in the odd edge case where both have data if we decide to support that).